listenbrainz/webserver/static/js/jsx/follow-users.jsx

```diff
@@ -51,14 +92,32 @@ export class FollowUsers extends React.Component {
       <div className="panel panel-primary">
         <div className="panel-heading">
           <i className="fas fa-sitemap fa-2x fa-flip-vertical"></i>
-          <span style={{ fontSize: "x-large", marginLeft: "0.55em", verticalAign: "middle" }}>
+          <span style={{fontSize: "x-large", marginLeft: "0.55em", verticalAign: "middle" }}>
             Follow users
-              </span>
+          </span>
+          <input type="text" placeholder={this.props.listName} style={{"float": "right"}}
+              ref={(input) => this.nameInput = input} />
+
+
         </div>
         <div className="panel-body">
-          <div className="text-muted">
-            Add a user to discover what they are listening to:
-              </div>
+          <div className="input-group">
+            <span className="text-muted">
+              Add a user to discover what they are listening to:
+            </span>
+            <span className="input-group-btn btn-primary">
+              <button className="btn btn-primary" type="button" onClick={this.newFollowList.bind(this)}>
+                <span className="fa fa-plus" aria-hidden="true"></span> New List
+              </button>
+            </span>
+            <div style={{"width": "1px", "height": "100%"}}/>
+            <span className="input-group-btn btn-primary">
+              <button className="btn btn-primary" type="button" onClick={this.saveFollowList.bind(this)}>
+                <span className="fa fa-save" aria-hidden="true"></span> Save
+              </button>
+            </span>
```

Comment:
UI-wise, I think the group input+New List+Save should appear on the right of what is currently the panel-body.

I would suggest putting two divs in panel-body, with `col-md-9` and `col-md-3` classes respectively, and arrange input+New List+Save in that second column.

Quick sketch:
<img width="1161" alt="capture d ecran 2019-03-01 a 12 24 29" src="https://user-images.githubusercontent.com/6179856/53635325-06f5c900-3c1d-11e9-8061-492e6dc5fc6d.png">

Reply:
We discussed that it'd be better for @MonkeyDo to do this in a new PR, given my great frontend skillz.